File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    static int answer = 0;
    static boolean[] visited; //방문 여부

    public int solution(int n, int[][] computers) {
        visited = new boolean[n];
        
        
        for(int curr=0; curr<n; curr++){
            if(!visited[curr]){
                answer++;
                dfs(curr, computers, n);
            }
        }
        
        return answer;
    }
    
    private void dfs(int curr, int[][] computers, int n){
        for(int nxt=0; nxt<n; nxt++){
            if(computers[curr][nxt]==1 && !visited[nxt]){
                visited[nxt] = true;
                dfs(nxt, computers, n);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
'''
영역 개수 구하기 문제
    answer = 0
    
    visited = [False for _ in range(n)]
    
    def dfs(curr):
        for nxt in range(n):
            if computers[curr][nxt]==1 and not visited[nxt]:
                visited[nxt] = True
                dfs(nxt)
    
    for curr in range(n):
        if not visited[curr]:
            answer+=1
            visited[curr] = True
            dfs(curr)
    
    return answer
'''

def solution(n, computers):
    answer = 0
    
    visited = [False for _ in range(n)];
    
    def dfs(curr):
        for nxt in range(n):
            if computers[curr][nxt]==1 and not visited[nxt]:
                visited[nxt] = True
                dfs(nxt)
    
    for curr in range(n):
        if not visited[curr]:
            answer+=1
            dfs(curr)
    
    return answer
    
    
